syntax = "proto3";
package customercenter;
//option java_multiple_files = true;
//option java_package = "com.szwdcloud.customercenter.proto.customercenter";
//option java_outer_classname = "CCCommonMessageProto";
// 引入外部的proto对象
import "google/protobuf/any.proto";

service CommonService {
    // 根据条件返回客户信息
    rpc listProvince (RequestListProvince) returns (AddressList){}
    rpc listCity (RequestListCity) returns (AddressList){}
    rpc listDistrict (RequestListDistrict) returns (AddressList){}

}
//分页封装数据
message CCBatchDataPage {
    /** 当前页 */
    int32 page = 1;
    /** 分页数 */
    int32 pagesize = 2;
    /** 开始行*/
    int32 offset = 3;
    /**总记录数*/
    float totalRecords = 4;
    /** 具体数据 */
    repeated google.protobuf.Any rows = 5; // 任何对象
}

//分页请求数据
message CCDataPageRequestParam {
    /** 当前页 */
    int32 page = 1;
    /** 分页数 */
    int32 pagesize = 2;
}

message AddressList{
    repeated AddressVo addressVo = 1;

}

//地址信息
message AddressVo{
    //县/区id
    string id = 1;
    ///省份id
    string provinceId = 2;
    //省份名称
    string provinceName = 3;
    //城市id
    string cityId = 4;
    //下辖城市名称
    string cityName = 5;
    //下辖县/区名称
    string districtName = 6;

}

///////////////////////////////////request message////////////////////////////////////
message RequestListProvince {
    string code = 1;
}

message RequestListCity {
    string provinceId = 1;
}

message RequestListDistrict {
    string cityId = 1;
}

///////////////////////////////////response message////////////////////////////////////